packages/stablestudio-plugin-comfyui/src/index.ts

```diff
@@ -0,0 +1,615 @@
+) => {
+  imageStrength = Math.max(0, Math.min(0.99, imageStrength));
+  return {
+    "3": {
+      class_type: "KSampler",
+      inputs: {
+        cfg: cfgScale,
+        denoise: 1 - imageStrength,
+        latent_image: ["17", 0],
+        model: ["4", 0],
+        negative: ["7", 0],
+        positive: ["6", 0],
+        sampler_name: sampler,
+        scheduler: "normal",
+        seed: seed,
+        steps: steps,
+      },
+    },
+    "4": {
+      class_type: "CheckpointLoaderSimple",
+      inputs: {
+        ckpt_name: model,
+      },
+    },
+    "6": {
+      class_type: "CLIPTextEncode",
+      inputs: {
+        clip: ["4", 1],
+        text: prompt,
+      },
+    },
+    "7": {
+      class_type: "CLIPTextEncode",
+      inputs: {
+        clip: ["4", 1],
+        text: negative_prompt,
+      },
+    },
+    "8": {
+      class_type: "VAEDecode",
+      inputs: {
+        samples: ["3", 0],
+        vae: ["4", 2],
+      },
+    },
+    "9": {
+      class_type: "SaveImage",
+      inputs: {
+        filename_prefix: "ComfyUI",
+        images: ["8", 0],
+      },
+    },
+    "10": {
+      class_type: "LoadImage",
+      inputs: {
+        image: imageFilename,
+        "choose file to upload": "image",
+      },
+    },
+    "17": {
+      class_type: "VAEEncodeForInpaint",
+      inputs: {
+        grow_mask_by: 0,
+        mask: ["21", 0],
+        pixels: ["10", 0],
+        vae: ["4", 2],
+      },
+    },
+    "20": {
+      class_type: "LoadImageMask",
+      inputs: {
+        channel: "red",
+        "choose file to upload": "image",
+        image: maskFilename,
+      },
+    },
+    "21": {
+      class_type: "InvertMask",
+      inputs: {
+        mask: ["20", 0],
+      },
+    },
+  };
+};
+
+const getExtras = () => {
+  const extras: StableStudio.StableDiffusionExtra[] = [];
+  const lora: StableStudio.StableDiffusionExtra = {
+    name: "Lora",
+    description: "Load LoRA to model",
+    type: "select",
+    data: {
+      type: "list",
+      data: ["lora1", "lora2", "lora3"],
+      default: "lora2",
+    },
+  };
+  const aitemplate: StableStudio.StableDiffusionExtra = {
+    name: "AITemplate",
+    description: "Accelerate inference with AITemplate",
+    type: "boolean",
+    data: {
+      type: "boolean",
+      default: false,
+      data: false,
+    },
+  };
+  const randomSlider: StableStudio.StableDiffusionExtra = {
+    name: "RandomSlider",
+    description: "Randomize the slider",
+    type: "slider",
+    data: {
+      type: "slider",
+      default: 0,
+      data: 50,
+      min: 0,
+      max: 100,
+      step: 1,
+    },
+  };
+  extras.push(lora);
+  extras.push(aitemplate);
+  extras.push(randomSlider);
+  return extras;
+};
+
+const getModels = async (apiUrl: string) => {
+  const response = await fetch(`${apiUrl}/object_info`);
+  if (response.ok) {
+    const json = await response.json();
+    return json.CheckpointLoaderSimple.input.required.ckpt_name[0];
```

Comment:
A similar request could be made to get specific model types from other `Loader's` input lists, VAE, LoRA etc